landscape/client/serviceconfig.py

```diff
@@ -0,0 +1,129 @@
+"""Interface for interacting with service managers managing landscape-client.
+
+This module exports a single class, `ServiceConfig`, which is aliased to a
+concrete implementation based on configuration. Concrete implementations
+provide the following interface:
+
+set_start_on_boot(flag: bool) -> None
+restart_landscape() -> None
+stop_landscape() -> None
+is_configured_to_run() -> bool
+
+Methods should raise `ServiceConfigException` if necessary.
+"""
+import os
+import subprocess
+
+SERVICE_NAME = "landscape-client"
+SYSTEMD_SERVICE = f"{SERVICE_NAME}.service"
+
+SNAPCTL = "/usr/bin/snapctl"
+SYSTEMCTL = "/usr/bin/systemctl"
+
+
+class ServiceConfigException(Exception):
+    """
+    Error modifying the landscape-client service's status or configuration.
+    """
+
+
+class SystemdConfig:
+    """
+    A collection of methods for driving the landscape-client systemd service.
+    """
+
+    @staticmethod
+    def set_start_on_boot(flag: bool) -> None:
+        action = "enable" if flag else "disable"
+        subprocess.run(
+            [SYSTEMCTL, action, SYSTEMD_SERVICE],
+            stdout=subprocess.PIPE,
+        )
+
+    @staticmethod
+    def restart_landscape() -> None:
+        try:
+            subprocess.run(
+                [SYSTEMCTL, "restart", SYSTEMD_SERVICE],
```

Comment:
This code is repeated alot. Could be in a helper function inside the class that calls subprocess.run with the necessary args. Same in the SnapConfig class